### PR TITLE
Content negotiation

### PR DESCRIPTION
**Pull request by deadtrickster**
Enable content negotiation and set config default format to `auto`. Set default Accept value to `text/plain`.

### CODE REVIEW

File: src/elli_prometheus.erl [hunk 1]
```diff
@@ -41,7 +41,7 @@
 handle(Req, _Config) ->
   Path = elli_prometheus_config:path(),
   case {elli_request:method(Req),elli_request:raw_path(Req)} of
-    {'GET', Path} -> format_metrics();
+    {'GET', Path} -> format_metrics(Req);
```
Comment (yurrriq) [inline src/elli_prometheus.erl:44]: Should `"text/plain"` be a binary?

**Comment by yurrriq** [comment]
Thanks for this. Looks great.

File: src/elli_prometheus.erl (hunk 1 above)
Comment (deadtrickster) [reply src/elli_prometheus.erl:44]: Wrong line?  :-)

Accept (the lib) works with binaries too.

**Comment by yurrriq** [comment]
Probably. I'm commenting on my phone. Looks like `elli:header()` allows `string()` `Value`s too, so this is fine. Thanks again!

https://github.com/elli-lib/elli/blob/2.0.0/src/elli.erl#L40

**Comment by yurrriq** [comment]
Actually, can you rebase on `develop` so we can merge it there? I'd like to keep `master` for releases only.

**Comment by yurrriq** [comment]
Perhaps I should make `develop` the default branch and write a little `CONTRIBUTING.md`.

**Comment by deadtrickster** [comment]
Develop is outdated here, Probably sync it with master first?

**Comment by yurrriq** [comment]
Will do. Gonna be later though, heading out for a bit now I think.

**Comment by deadtrickster** [comment]
ping

**Comment by yurrriq** [comment]
Thanks for your patience :). I've synced `develop` with `master` now.

**Comment by deadtrickster** [comment]
changed based branch to elli-lib:develop. can we merge now?

**Pull request merged**